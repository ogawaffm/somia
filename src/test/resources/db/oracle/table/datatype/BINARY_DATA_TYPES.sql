BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.BINARY_DATA_TYPES PURGE';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
CREATE TABLE TEST.BINARY_DATA_TYPES
(
    RAW_1_COL    RAW(1),
    RAW_2000_COL RAW(2000),

    LONG_RAW_COL LONG RAW,
    BLOB_COL     BLOB

)
