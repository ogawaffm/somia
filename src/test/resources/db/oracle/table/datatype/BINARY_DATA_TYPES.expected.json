{
  "name": "BINARY_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "RAW_1_COL",
      "type": {
        "name": "RAW",
        "typeNumber": -3,
        "className": "byte[]",
        "precision": 1,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 2,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "RAW_2000_COL",
      "type": {
        "name": "RAW",
        "typeNumber": -3,
        "className": "byte[]",
        "precision": 2000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 4000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "LONG_RAW_COL",
      "type": {
        "name": "LONG RAW",
        "typeNumber": -4,
        "className": "byte[]",
        "precision": 0,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 65535,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "BLOB_COL",
      "type": {
        "name": "BLOB",
        "typeNumber": 2004,
        "className": "java.sql.Blob",
        "precision": 4000,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": false,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 8000,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}